src/services/confirmation.statement.service.ts

```diff
@@ -0,0 +1,33 @@
+import { createPrivateApiClient } from "private-api-sdk-node";
+import {Session} from "@companieshouse/node-session-handler";
+import {SignInInfoKeys} from "@companieshouse/node-session-handler/lib/session/keys/SignInInfoKeys";
+import {AccessTokenKeys} from "@companieshouse/node-session-handler/lib/session/keys/AccessTokenKeys";
+import {SessionKey} from "@companieshouse/node-session-handler/lib/session/keys/SessionKey";
+import logger from "../utils/logger";
+import PrivateApiClient from "private-api-sdk-node/dist/client";
+import {CHS_API_KEY, INTERNAL_API_URL} from "../utils/properties";
+import {ConfirmationStatementService} from "private-api-sdk-node/dist/services/confirmation-statement";
+import Resource from "@companieshouse/api-sdk-node/dist/services/resource";
+
+export const createConfirmationStatement = async (session: Session,
+                                                  transactionId: string): Promise<Resource<any>> => {
+    const client = createOAuthApiClient(session);
+    const csService: ConfirmationStatementService = client.confirmationStatementService;
+    return await csService.postNewConfirmationStatement(transactionId);
```

Comment:
the csService.postNewConfirmationStatement is missing a return type on the function. It returns Resource<ConfirmationStatementSubmission> so this createConfirmationStatement function should return Promise<Resource<ConfirmationStatementSubmission>> instead of Promise<Resource<any>>

Reply:
That's the othe rmethod in the  cs privatesdk class "postConfirmationStatementSubmission" you're thinking of.

Reply:
that's the one - https://github.com/companieshouse/private-api-sdk-node/blob/master/src/services/confirmation-statement/service.ts#L33
The return type needs to be added to the private sdk on the function https://github.com/companieshouse/private-api-sdk-node/blob/master/src/services/confirmation-statement/service.ts#L33. Then we can return a `Resource<ConfirmationStatementSubmission>` from the function you are adding now instead of `any`

Reply:
right i think i am looking at the wrong one

Reply:
You are, but that's a good point about the missing return type for submission but that'll need to be changed in private sdk node.

Reply:
ok, we can look at that separately